orca-sql/src/main/kotlin/com/netflix/spinnaker/orca/sql/pipeline/persistence/SqlExecutionRepository.kt

```diff
@@ -427,6 +451,129 @@ class SqlExecutionRepository(
+      .and(field("config_id").`in`(*pipelineConfigIds.toTypedArray()))
+
+    var table = if (jooq.dialect() == SQLDialect.MYSQL) PIPELINE.tableName.forceIndex("pipeline_application_idx")
+    else PIPELINE.tableName
+    // baseQueryPredicate for the flow with statuses
+    if (criteria.statuses.isNotEmpty() && criteria.statuses.size != ExecutionStatus.values().size) {
+      val statusStrings = criteria.statuses.map { it.toString() }
+      baseQueryPredicate = baseQueryPredicate
+        .and(field("status").`in`(*statusStrings.toTypedArray()))
+
+      table = if (jooq.dialect() == SQLDialect.MYSQL) PIPELINE.tableName.forceIndex("pipeline_application_status_starttime_idx")
+      else PIPELINE.tableName
+    }
+
+    val finalResult: MutableList<String> = mutableListOf()
+
+    withPool(poolName) {
+      val baseQuery = jooq.select(field("config_id"), field("id"))
+        .from(table)
+        .where(baseQueryPredicate)
+         .orderBy(field("config_id"))
+         .fetch().intoGroups("config_id", "id")
+
+        baseQuery.forEach {
+          val count = it.value.size
+          if (criteria.pageSize < count) {
+            finalResult.addAll(it.value
+              .stream()
+              .skip((count - criteria.pageSize).toLong())
+              .collect(toList()) as List<String>
+            )
+          } else {
+            finalResult.addAll(it.value as List<String>)
+          }
+        }
+    }
+    return finalResult
+  }
+
+  /**
+   * It executes the following query to get execution details for n executions at a time in a specific application
+   *
+   * SELECT id, body, compressed_body, compression_type, `partition`
+       FROM pipelines force index (`pipeline_application_idx`)
+       left outer join
+       pipelines_compressed_executions
+       using (`id`)
+       WHERE (
+         application = "<myapp>" and
+         id in ('id1', 'id2', 'id3')
+       );
+   *
+   * it then gets all the stage information for all the executions returned from the above query.
+   */
+  override fun retrievePipelineExecutionDetailsForApplication(
+    application: String,
+    pipelineExecutions: List<String>,
+    queryTimeoutSeconds: Int
+  ): Collection<PipelineExecution> {
+    withPool(poolName) {
+      val baseQuery = jooq.select(selectExecutionFields(compressionProperties))
+        .from(
+          if (jooq.dialect() == SQLDialect.MYSQL) PIPELINE.tableName.forceIndex("pipeline_application_idx")
+          else PIPELINE.tableName
+        )
+        .leftOuterJoin(PIPELINE.tableName.compressedExecTable).using(field("id"))
+        .where(
+          field("application").eq(application)
+            .and(field("id").`in`(*pipelineExecutions.toTypedArray()))
+        )
+        .queryTimeout(queryTimeoutSeconds) // add an explicit timeout so that the query doesn't run forever
+        .fetch()
+
+      log.info("getting stage information for all the executions found so far")
```

Comment:
This doesn't seem valuable enough to leave at info.
```suggestion
      log.debug("getting stage information for all the executions found so far")
```